helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -210,6 +213,18 @@ private void updateStateTransitionMessageThreadPool(Message message, HelixManage
       return;
     }
 
+    if (!_batchMessageThreadpoolChecked) {
```

Comment:
This is not a proper solution for the problem. We shall keep the backward compatible. HelixTaskExecutor has a default constructor now. In this case, if the HelixTaskExecutor instantiated by default constructor does not have any information for configurable threadpool. That's why there is also a logic to make the threadpool configurable in a separate method (not only for batch message thread pool). Otherwise, the user defined the information will be ignored. It is not good just remove the public constructor.